src/user.rs

```diff
@@ -200,14 +189,14 @@ pub async fn user_main(
 async fn invoke_task_function(
     function: &GooseTaskFunction,
     thread_user: &GooseUser,
-    task_index: usize,
+    thread_task_index: usize,
```

Comment:
Though not strictly required, it was odd how all other variables were prefixed with `thread_` -- this change improves the readability of the code affected by these changes.